sdks/java/harness/src/test/java/org/apache/beam/fn/harness/CachesTest.java

```diff
@@ -33,32 +34,53 @@
 @RunWith(JUnit4.class)
 public class CachesTest {
   @Test
-  public void testNoopCache() {
+  public void testNoopCache() throws Exception {
     Cache<String, String> cache = Caches.noop();
     cache.put("key", "value");
     assertNull(cache.peek("key"));
     assertEquals("value", cache.computeIfAbsent("key", (unused) -> "value"));
     assertNull(cache.peek("key"));
-    assertThat(cache.keys(), is(emptyIterable()));
   }
 
   @Test
-  public void testEternalCache() {
+  public void testShrinkableIsShrunk() throws Exception {
+    Shrinkable<Object> shrinkable =
+        new Shrinkable<Object>() {
+
+          @Override
+          public Object shrink() {
+            return "wasShrunk";
+          }
+        };
+
+    Cache<Object, Object> cache =
+        Caches.forCache(new ShrinkOnEviction(CacheBuilder.newBuilder().maximumSize(1)).getCache());
```

Comment:
nit: would it be possible to create this through a public API rather than opening up `forCache`? It used to be private

Reply:
There isn't a way to make the ShrinkOnEviction have a final field storing a reference to the cache that I know of.

forCache is still package private, I'll mark it as @VisibleForTesting in a follow-up PR.